lib/mkl/interfaces.jl

```diff
@@ -31,3 +42,233 @@ end
 function LinearAlgebra.generic_trimatdiv!(C::oneMatrix{T}, uploc, isunitc, tfun::Function, A::oneSparseMatrixCSR{T}, B::oneMatrix{T}) where T <: BlasFloat
     sparse_trsm!(uploc, tfun === identity ? 'N' : tfun === transpose ? 'T' : 'C', 'N', isunitc, one(T), A, B, C)
 end
+
+# Handle Transpose and Adjoint wrappers for sparse matrices
+# Let the low-level wrappers handle the CSC->CSR conversion and flip_trans logic
+
+# Matrix-vector multiplication with transpose/adjoint
+function LinearAlgebra.generic_matvecmul!(C::oneVector{T}, tA::AbstractChar, A::Transpose{T, <:oneSparseMatrixCSR{T}}, B::oneVector{T}, _add::MulAddMul) where {T <: BlasFloat}
+    tA = tA in ('S', 's', 'H', 'h') ? 'N' : tA
+    tA_final = tA == 'N' ? 'T' : (tA == 'T' ? 'N' : 'C')
```

Comment:
@michel2323 In the case `tA = 'C'`, you need to do a product with the conjugate.

Reply:
I am wondering if you didn't mixed `BlasReal` with `BlasFloat`?